shared/chat/conversation/info-panel/add-to-channel/index.new.tsx

```diff
@@ -0,0 +1,201 @@
+import * as React from 'react'
+import * as Kb from '../../../../common-adapters'
+import * as Styles from '../../../../styles'
+import * as ChatConstants from '../../../../constants/chat2'
+import * as ChatTypes from '../../../../constants/types/chat2'
+import * as Chat2Gen from '../../../../actions/chat2-gen'
+import * as TeamConstants from '../../../../constants/teams'
+import * as TeamTypes from '../../../../constants/types/teams'
+import * as Container from '../../../../util/container'
+import {useTeamDetailsSubscribe} from '../../../../teams/subscriber'
+import {pluralize} from '../../../../util/string'
+import {memoize} from '../../../../util/memoize'
+
+type Props = Container.RouteProps<{
+  conversationIDKey: ChatTypes.ConversationIDKey
+}>
+
+const sortMembers = memoize((members: TeamTypes.TeamDetails['members']) =>
+  [...members.values()]
+    .filter(m => m.type !== 'restrictedbot' && m.type !== 'bot')
+    .sort((a, b) => a.username.localeCompare(b.username))
+)
+
+const AddToChannel = (props: Props) => {
+  const conversationIDKey = Container.getRouteProps(
+    props,
+    'conversationIDKey',
+    ChatConstants.noConversationIDKey
+  )
+  const dispatch = Container.useDispatch()
+  const nav = Container.useSafeNavigation()
+
+  const [toAdd, setToAdd] = React.useState<Set<string>>(new Set())
+  const [filter, setFilter] = React.useState('')
+  const fLower = filter.toLowerCase()
+
+  const {channelname, teamname, teamID} = Container.useSelector(s =>
+    ChatConstants.getMeta(s, conversationIDKey)
+  )
+  const participants = Container.useSelector(s => ChatConstants.getParticipantInfo(s, conversationIDKey)).all
+  const teamDetails = Container.useSelector(s => TeamConstants.getTeamDetails(s, teamID))
+  const allMembers = sortMembers(teamDetails.members)
+  const membersFiltered = allMembers.filter(
+    m => m.username.toLowerCase().includes(fLower) || m.fullName.toLowerCase().includes(fLower)
+  )
```

Comment:
This seems like it'll be brutal on large teams--it'll have to recalculate on each filtering character. If we stored the `membersFiltered` in the state, adding a character could simply use the previous filtered set, and we'd only have to recalculate from `allMembers` when the user removes a filter character or changes it altogether. So, if the previous filter is a prefix of the new filter, we can reuse the previous `membersFiltered`.

Reply:
Even more fancy, we could consider using something like this: https://www.npmjs.com/package/trie-prefix-tree

Reply:
From testing locally on keybasefriends it doesn't cause problems for me. I agree it's doing extra work but think we can leave optimization for a future pass.